sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -45,11 +45,9 @@ import org.apache.spark.util.random.RandomSampler
  * The AstBuilder converts an ANTLR4 ParseTree into a catalyst Expression, LogicalPlan or
  * TableIdentifier.
  */
-class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging {
+class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with Logging {
```

Comment:
Let me try it later. It sounds like they are stateless. 

`CatalystSqlParser` was originally an object when @hvanhovell replaced ANTLR4 for ANTLR3. `SparkSqlAstBuilder` and `AstBuilder` are a class since day one.